staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -263,7 +263,7 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 		oldModifyResponse := proxy.ModifyResponse
 		proxy.ModifyResponse = func(response *http.Response) error {
 			code := response.StatusCode
-			if code >= 300 && code <= 399 {
+			if code >= 300 && code <= 399 && len(response.Header.Get("Location")) > 0 {
```

Comment:
`response.Header.Get` returns empty string if the header is missing right?

Reply:
yup, and is case-insensitive - https://pkg.go.dev/net/http#Header.Get

Reply:
looks like it! https://pkg.go.dev/net/http#Header.Get